services/bundle_analysis/new_notify/messages/commit_status.py

```diff
@@ -0,0 +1,128 @@
+from services.notification.notifiers.base import NotificationResult
+
+log = logging.getLogger(__name__)
+
+
+class BundleCommentTemplateContext(TypedDict):
+    prefix_message: str
+    change_readable: str
+    warning_threshold_readable: str
+
+
+class CommitStatusMessageStrategy(MessageStrategyInterface):
+    def build_message(self, context: CommitStatusNotificationContext) -> str | bytes:
+        template = loader.get_template(
+            "bundle_analysis_notify/commit_status_summary.md"
+        )
+        # Prefix message is based on the commit status level
+        prefix_message = {
+            CommitStatusLevel.INFO: "",
+            CommitStatusLevel.WARNING: "Passed with Warnings -",
+            CommitStatusLevel.ERROR: "Failed -",
+        }.get(context.commit_status_level)
+
+        warning_threshold = context.user_config.warning_threshold
+        if warning_threshold.type == "absolute":
+            warning_threshold_readable = bytes_readable(warning_threshold.threshold)
+            absolute_change = context.bundle_analysis_comparison.total_size_delta
+            is_negative = absolute_change < 0
+            change_readable = ("-" if is_negative else "") + bytes_readable(
+                absolute_change
+            )
+        else:
+            warning_threshold_readable = str(warning_threshold.threshold) + "%"
+            change_readable = (
+                str(context.bundle_analysis_comparison.percentage_delta) + "%"
+            )
+
+        context = BundleCommentTemplateContext(
+            prefix_message=prefix_message,
+            change_readable=change_readable,
+            warning_threshold_readable=warning_threshold_readable,
+        )
+        return template.render(context)
+
+    def _cache_key(self, context: CommitStatusNotificationContext) -> str:
+        return make_hash_sha256(
+            dict(
+                type="status_check_notification",
+                repoid=context.repository.repoid,
+                base_commitid=context.base_commit.commitid,
+                head_commitid=context.commit.commitid,
+                notifier_name="bundle_analysis_commit_status",
+                notifier_title="codecov/bundles",
+            )
+        )
+
+    def _is_message_unchanged(
+        self, context: CommitStatusNotificationContext, message: str | bytes
+    ) -> bool:
+        """Checks if the message we are about to change was already sent."""
+        last_payload = cache.get_backend().get(self._cache_key(context))
+        print("CACHE", cache)
+        print("LAST PAYLOAD", self._cache_key(context), last_payload)
+        if last_payload is NO_VALUE or last_payload != message:
+            return False
+        return True
+
+    async def send_message(
+        self, context: CommitStatusNotificationContext, message: str | bytes
+    ) -> NotificationResult:
+        repository_service = context.repository_service
+        if not self._is_message_unchanged(context, message):
```

Comment:
I would actually inline the `_is_message_unchanged` call here, as I believe its the only callsite? That would allow you to only do the (potentially expensive) `make_hash_sha256` call only once, maybe also inlining the `_cache_key` method.

And I’m personally a big fan on early returns, so I would turn this `if` around, and return the `False` result right away, and then unindent the rest of the block where you actually set the cache.